### PR TITLE
fix(dashboard): add toast if JSON metadata is invalid

### PR DESCRIPTION
**Pull request by stephenLYZ**
### SUMMARY

When the json metadata of the dashboard is not legal, the user clicks the save button without any response and no other prompt, this PR adds toast to inform the user.

### before

https://user-images.githubusercontent.com/11830681/180412795-4eae920a-6ce2-4d55-b209-98baedb42448.mov

### after

https://user-images.githubusercontent.com/11830681/180412783-b5464423-d9a6-42ea-acea-c10a8116d545.mov

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Review by zhaoyongjie — APPROVED** [review]
LGTM

**Review by Antonio-RiveroMartnez — APPROVED** [review]

File: superset-frontend/src/dashboard/components/PropertiesModal/index.tsx [hunk 1]
```diff
@@ -294,7 +296,12 @@ const PropertiesModal = ({
 
     // color scheme in json metadata has precedence over selection
     if (currentJsonMetadata?.length) {
-      const metadata = JSON.parse(currentJsonMetadata);
+      let metadata;
+      try {
+        metadata = JSON.parse(currentJsonMetadata);
+      } catch (error) {
+        addDangerToast(t('JSON metadata is invalid!'));
```
Comment (etr2460) [inline superset-frontend/src/dashboard/components/PropertiesModal/index.tsx:303]: @zhaoyongjie shouldn't this re-throw the error (or at least return from the function) if we don't have valid metadata here so we don't try to save an invalid payload?

Comment (zhaoyongjie) [reply superset-frontend/src/dashboard/components/PropertiesModal/index.tsx:303]: @etr2460 good point, let's try to make an invalid JSON to verify.

Comment (john-bodley) [reply superset-frontend/src/dashboard/components/PropertiesModal/index.tsx:303]: @zhaoyongjie is there any update regarding a fix for this?

Comment (zhaoyongjie) [reply superset-frontend/src/dashboard/components/PropertiesModal/index.tsx:303]: @john-bodley @etr2460 add a new PR at [here](https://github.com/apache/superset/pull/21524).

**Pull request merged**